src/sentry/event_manager.py

```diff
@@ -523,8 +522,6 @@ def save(self, project_id, raw=False, assume_normalized=False):
         # We need to swap out the data with the one internal to the newly
         # created event object
         event = self._get_event_instance(project_id=project_id)
-        self._data = data = event.data.data
```

Comment:
I am quite concerned about this line.
My comment on the previous version (where this became an assertion) was about using only one of the three in the rest of the method, but there is still code that writes and reads from both data and event.data.data (see event.get_hashes). 
Do you know why this line existed in the first place ?
In the new implementation self._data and event.data.data are not the same. self._data is a CanonicalKeyDict while event.data.data is an EventDict (which is a subclass). Is this intended ?